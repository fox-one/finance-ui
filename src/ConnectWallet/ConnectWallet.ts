import {
  Component,
  Vue,
  Prop
} from 'vue-property-decorator';
import classnames from '@utils/classnames';
import { $t } from '@utils/t';
/* import types */
import type { CreateElement, VNode } from 'vue';

@Component
export class ConnectWallet extends Vue {
  @Prop({ type: String, default: '' }) private className!: string;
  @Prop({ type: String, default: '' }) private text!: string;

  public render(h: CreateElement): VNode {
    const classes = classnames('connect-wallet');
    return h(
      'div',
      {
        staticClass: classes(),
        class: this.className
      },
      [
        h(
          'span',
          {
            staticClass: classes('text', 'f-greyscale-1 f-caption font-weight-bold pr-4'),
            on: {
              ...this.$listeners
            }
          },
          [this.text || $t(this, 'connect_wallet')]
        )
      ]
    );
  }
}

export default ConnectWallet;