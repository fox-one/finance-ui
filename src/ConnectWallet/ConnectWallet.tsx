import {
  defineComponent,
  onMounted
} from '@vue/composition-api';
import classnames from '@utils/classnames';
/* import types */
import type { CreateElement, VNode } from 'vue';

export default defineComponent({
  name: 'ConnectWallet',
  props: {
    text: {
      type: [String, Object],
      default: 'Connect Wallet'
    }
  },
  setup() {
    const classes = classnames('connect-wallet');
    onMounted(() => {
      console.info('ConnectWallet mounted!');
    });

    return { classes };
  },
  render(h: CreateElement): VNode {
    return (
      <div class={this.classes()}>
        <span class={this.classes('text', 'f-greyscale-1 f-caption font-weight-bold pr-4')}>
          { this.text }
        </span>
      </div>
    );
  }
});