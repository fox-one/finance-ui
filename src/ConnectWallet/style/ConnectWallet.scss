@import '../../styles/variable.scss';

.#{$prefix}-connect-wallet {
  &-text {
    background-image: url('../../assets/images/svg/connect.svg');
    background-repeat: no-repeat;
    background-position: right center;
    background-size: 12px;
    cursor: pointer;
  }
}

.theme--dark {
  .#{$prefix}-connect-wallet {
    &-text {
      background-image: url('../../assets/images/svg/connect-dark.svg');
    }
  }
}