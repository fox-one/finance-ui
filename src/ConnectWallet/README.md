# ConnectWallet

## Example

```vue
<template>
  <connect-wallet />
</template>

<script lang="ts">
  import { Component, Vue } from 'vue-property-decorator';
  import { ConnectWallet } from './ConnectWallet';

  @Component(
    components: {
      ConnectWallet
    }
  )
  export class App extends Vue {
    mounted () {
      console.info('mounted!')
    }
  }
</script>
```
