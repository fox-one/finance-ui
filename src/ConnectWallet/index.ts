import ConnectWallet from './ConnectWallet';

export { ConnectWallet } from './ConnectWallet';

export default ConnectWallet;
