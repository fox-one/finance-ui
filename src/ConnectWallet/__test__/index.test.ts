import { shallowMount, mount } from '@vue/test-utils';
import ConnectWallet from '../';

describe('ConnectWallet', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(ConnectWallet);
    expect(wrapper).toMatchSnapshot();
  });
});
