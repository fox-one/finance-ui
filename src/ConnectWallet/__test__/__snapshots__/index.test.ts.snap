// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectWallet renders correctly 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
